plugins/ccgx/fu-ccgx-hpi.c

```diff
@@ -0,0 +1,1622 @@
+static gboolean
+fu_ccgx_hpi_read_flash (FuDevice *device, CyHPIHandle *handle, guint16 row_num, GError **error)
+{
+	guint8 flash_cmd[4];
+	flash_cmd[0] = CY_PD_FLASH_READ_WRITE_CMD_SIG;
+	flash_cmd[1] = CY_PD_REG_FLASH_ROW_READ_CMD;
+	flash_cmd[2] = row_num & 0xFF;
+	flash_cmd[3] = (row_num >> 8);
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	if (!fu_ccgx_hpi_reg_write(device, &handle->i2c_handle, handle->hpi_addr_byte, CY_PD_REG_FLASH_READ_WRITE_ADDR, flash_cmd, 4, error)) {
+		g_prefix_error (error, "read flash error: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+/**
+ * fu_ccgx_hpi_validate_fw:
+ * @device: #FuDevice
+ * @handle: HPI handle
+ * @fw_index firmware index (FW1 or FW2)
+ * @error: a #GError or %NULL
+ *
+ * Validate firmware written
+ *
+ * Returns: %TRUE for success
+*/
+static gboolean
+fu_ccgx_hpi_validate_fw (FuDevice *device,CyHPIHandle *handle, guint8 fw_index, GError **error)
+{
+	g_return_val_if_fail (handle != NULL, FALSE);
+	if (!fu_ccgx_hpi_reg_write(device, &handle->i2c_handle, handle->hpi_addr_byte, CY_PD_REG_VALIDATE_FW_ADDR, &fw_index, 1, error)) {
+		g_prefix_error (error, "validate fw error: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+/**
+ * fu_ccgx_hpi_reset_device:
+ * @device: #FuDevice
+ * @handle: HPI handle
+ * @error: a #GError or %NULL
+ *
+ * Reset device
+ *
+ * Returns: %TRUE for success
+*/
+static gboolean
+fu_ccgx_hpi_reset_device (FuDevice *device, CyHPIHandle *handle, GError **error)
+{
+	guint8 reset_cmd[2];
+	reset_cmd[0] = CY_PD_DEVICE_RESET_CMD_SIG;
+	reset_cmd[1] = CY_PD_REG_RESET_DEVICE_CMD;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	if (!fu_ccgx_hpi_reg_write_no_resp(device, &handle->i2c_handle, handle->hpi_addr_byte, CY_PD_REG_RESET_ADDR, reset_cmd, 2, error)) {
+		g_prefix_error (error, "reset device error: ");
+		return FALSE;
+	}
+	return TRUE;
+}
+
+/**
+ * fu_ccgx_hpi_get_event:
+ * @device: #FuDevice
+ * @handle: HPI handle
+ * @reg_section register section type. see HPIRegSection
+ * @event[out] event response from device
+ * @io_timeout timeout to wait event
+ * @error: a #GError or %NULL
+ *
+ * Get event from device within specificed timeout
+ *
+ * Returns: %TRUE for success
+*/
+static gboolean
+fu_ccgx_hpi_get_event (FuDevice *device, CyHPIHandle *handle, HPIRegSection reg_section, CyPDResp*	event, guint32 io_timeout, GError **error )
+{
+	HPIEvent event_array[HPI_REG_SECTION_ALL + 1] = {0};
+	guint32 timeout_ms = io_timeout;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	g_return_val_if_fail (event != NULL, FALSE);
+	if (fu_ccgx_hpi_wait_for_event(device,handle,reg_section,event_array,timeout_ms,error)) {
+		*event = event_array[reg_section].event_code;
+		return TRUE;
+	}
+	return FALSE;
+}
+
+/**
+ * fu_ccgx_hpi_clear_all_event:
+ * @device: #FuDevice
+ * @handle: HPI handle
+ * @io_timeout timeout to clear event
+ * @error: a #GError or %NULL
+ *
+ * Clear all event from device
+ *
+*/
+static void
+fu_ccgx_hpi_clear_all_event (FuDevice *device,
```

Comment:
plural?

Reply:
Do you mean changing function name like below?
 fu_ccgx_hpi_clear_all_event -> fu_ccgx_hpi_clear_all_events